canonical-web-and-design/maas-ui
# feat(ui): migrate to redux-first-history

## Done

- Replace connected-react-router with redux-first-history to support React Router v6.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Load the app.
- Open your redux dev tools.
- In state check that the router state includes your current path.
- Navigate around the app and check that the path updates.

## Fixes

Fixes: #3505.
